test/extensions/filters/network/thrift_proxy/conn_manager_test.cc

```diff
@@ -74,7 +74,7 @@ class ThriftConnectionManagerTest : public testing::Test {
     decoder_filter_.reset(new NiceMock<ThriftFilters::MockDecoderFilter>());
     config_.reset(new TestConfigImpl(proto_config_, context_, decoder_filter_, stats_));
 
-    filter_.reset(new ConnectionManager(*config_));
+    filter_.reset(new ConnectionManager(*config_, random_));
```

Comment:
Because I don't think you care about the value in these tests, I think you could put
`ON_CALL(random_, random()).WillByDefault(Return(42))` here and remove the EXPECT_CALL lines. Or if that doesn't work, `EXPECT_CALL(random_, random()).Times(AnyNumber()).WillOnce(Return(42))`.

Reply:
didn't know about `WillByDefault` -- thank you.